extensions/Worker.Extensions.Storage.Blobs/src/BlobStorageConverter.cs

```diff
@@ -57,136 +59,166 @@ private async ValueTask<ConversionResult> ConvertFromBindingDataAsync(ConverterC
-                string msg = String.Format(CultureInfo.CurrentCulture,
-                    @"Binding parameters to complex objects uses JSON serialization.
-                    1. Bind the parameter type as 'string' instead to get the raw values and avoid JSON deserialization, or
-                    2. Change the blob to be valid json.
-                    The JSON parser failed: {0}",
-                    ex.Message);
+                return false;
+            }
 
-                return ConversionResult.Failed(new InvalidOperationException(msg, ex));
+            if (!(type.IsArray || typeof(IEnumerable).IsAssignableFrom(type)))
+            {
+                return false;
             }
-            catch (Exception ex)
+
+            if (string.IsNullOrEmpty(blobName))
             {
-                return ConversionResult.Failed(ex);
+                return true;
             }
-        }
 
-        private Dictionary<string, string> GetBindingDataContent(ModelBindingData bindingData)
-        {
-            return bindingData?.ContentType switch
+            if (BlobIsFileRegex.IsMatch(blobName))
             {
-                Constants.JsonContentType => new Dictionary<string, string>(bindingData?.Content?.ToObjectFromJson<Dictionary<string, string>>(), StringComparer.OrdinalIgnoreCase),
-                _ => throw new InvalidContentTypeException(Constants.JsonContentType)
-            };
+                throw new InvalidOperationException("Collections are not supported when binding to a specific blob file.");
+            }
+
+            return true;
         }
 
-        private async Task<object?> ConvertModelBindingDataAsync(IDictionary<string, string> content, Type targetType, ModelBindingData bindingData)
+        private async Task<object> BindToCollectionAsync(Type targetType, BlobContainerClient container, string blobPath)
         {
-            content.TryGetValue(Constants.Connection, out var connectionName);
-            content.TryGetValue(Constants.ContainerName, out var containerName);
-            content.TryGetValue(Constants.BlobName, out var blobName);
+            Type elementType = targetType.IsArray ? targetType.GetElementType() : targetType.GenericTypeArguments[0];
+
+            if (elementType == typeof(BlobContainerClient))
+            {
+                throw new InvalidOperationException("Collections of BlobContainerClient are not supported.");
+            }
 
-            if (string.IsNullOrEmpty(connectionName))
+            var resultType = typeof(List<>).MakeGenericType(elementType);
+            var result = (IList)Activator.CreateInstance(resultType);
+
+            await foreach (BlobItem blobItem in container.GetBlobsAsync(prefix: blobPath).ConfigureAwait(false))
             {
-                throw new ArgumentNullException(nameof(connectionName));
+                var element = await ToTargetTypeAsync(elementType, container, blobItem.Name);
+
+                if (element is not null)
```

Comment:
Should we be ignoring `null` values here? Are you expecting the `null` cases here to be a result of deserialization? If so, shouldn't those results still be provided/exposed?